datadog/agent/install.go

```diff
@@ -89,11 +87,13 @@ func updateAgentConfig(
 	agentConfigFullPath := path.Join(os.GetAgentConfigFolder(), "datadog.yaml")
 	var err error
 	var parts = []string{agentConfig}
+
+	for _, extraConfig := range extraAgentConfig {
+		agentConfig = agentConfig + "\n" + extraConfig
+	}
+
 	if agentConfig != "" {
-		agentConfigWithAPIKEY := env.AgentAPIKey().ApplyT(func(apiKey string) pulumi.String {
-			config := strings.ReplaceAll(agentConfig, "{{API_KEY}}", apiKey)
-			return pulumi.String(config)
-		}).(pulumi.StringInput)
+		agentConfigWithAPIKEY := pulumi.Sprintf("api_key: %v\n%v", env.AgentAPIKey(), agentConfig)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Is `env.AgentAPIKey()` safely checked and mandatory ? If empty, it would break the agent config load

Reply:
What happens if `api_key` is already there ?

Reply:
`env.AgentAPIKey()` is [mandatory](https://github.com/DataDog/test-infra-definitions/blob/67d9b1bee16a4e26ae2ee0d786e8047e85e0962f/common/config/environment.go#L97)

> What happens if api_key is already there

The Agent takes the second api key (the one provided by the user) and displays a warning.

Note: From datadog-agent repository, you will not have an easy way to have the API KEY.